config/k8s/wait.py

```diff
@@ -51,6 +51,12 @@ def find_pod(core, args):
                                     persist_config=False)
             core = client.CoreV1Api()
             continue
+        # exception might happen deep inside kubernetes, example:
+        # https://github.com/kernelci/kernelci-core/issues/1470#issue-1408983064
+        except Exception as e:
+            print(f"ERROR: find_pod generic exception: {e!r}")
+            continue
```

Comment:
How are we going to notice exceptions if we just catch them?  Now we need to dig into the logs to find them.  Is it really better to hide the problem than make it crash and then we can investigate what actually caused it in Kubernetes?

Reply:
As explained in comment - sometimes exceptions are just temporary networking errors to Kubernetes API or glitch inside Kubernetes node, so by catching exception and just printing it, we are giving find_pod chance to retry and recover. If it is really critical and repeating, it will try 10 times then just return None and whole script will fail, leading to job failure (then logs will be reviewed).